docs/architecture/adr-004-application-links-expiration.md

```diff
@@ -0,0 +1,80 @@
+other blockchains accounts. By doing this, a user can be verified as the owner of those accounts and prove to the system
+that they're not impersonating anyone else. This verification however remains valid only if the user
+never trades or sells their centralized-app username to someone else. If they do, the link to such username MUST be invalidated. 
+Unfortunately for us, it's not possible to understand when this happens since it's off-chain. 
+To prevent this situation, an "expiration time" SHOULD be added to the `ApplicationLink` object.
+
+## Decision
+
+To implement the link expiration we will act as follow:
+1) Extend the `ApplicationLink` structure by adding an `expiration_time` field that represent the time when the link will expire and will be
+   deleted from the store.
+2) Save a reference of the expiring link inside the store under the prefix `ExpiringApplicationLinkPrefix`. It will be formed by a prefix and a 
+   `timestamp` value which will make it easy to iterate over the expired links
+3) Add a new keeper method that will allow to iterate over these expired links
+4) Use the created iterator to implement another keeper method `DeleteExpiredApplicationLinks` that will take care of deleting all the expired
+   links
+5) Add this method to the `BeginBlock` function in order to trigger the deletion of expired link at each block
+
+We will also need to introduce a new `x/profiles` on chain parameter, `AppLinkParams`, which will register the default expiration time
+of all the app links and could be edited by the community with a governance proposal. 
+The parameter will be later used inside `StartProfileConnection` to calculated the estimated expire time of each `AppLink`.
+
+We also need to add a new `AppLinksParams` parameter to the `x/profiles` params, which will contain
+an `expiration_time` timestamp that will be added to all the `AppLinks` before they're stored in the chain. 
+The implementation for the new parameter is the following:
+
+## Consequences
+
+### Backwards Compatibility
+
+This update will affect the `ApplicationLink` object by adding the new `ExpirationBlockHeight` 
+field on it breaking the compatibility with the previous versions of the software. To allow
+a smooth update and overcome these compatibility issues, we need to set up a proper migration
+from the previous versions to the one that will include the additions contained in this ADR.
+
+### Positive
+
+* Considerably reduce the possibility of impersonation of entities and users of centralized apps;
+
+### Negative
+
+By adding the extra `expiration_time` field we are going to raise the overall `AppLinks` handling complexity.
+Since we're performing an iteration over all the expired link references inside at the start of each block this can require
+an amount of time that SHOULD be studied with benchmark tests during the implementation.
+
+### Neutral
+
+(none known)
+
+## Further Discussions
+
+## Test Cases [optional]
```

Comment:
We should add a test here to make sure that expired application links are deleted when necessary